beacon-chain/sync/pending_attestations_queue.go

```diff
@@ -99,7 +99,9 @@ func (s *Service) processPendingAtts(ctx context.Context) error {
 
 			// Start with a random peer to query, but choose the first peer in our unsorted list that claims to
 			// have a head slot newer or equal to the pending attestation's target boundary slot.
-			pid := pids[rand.Int()%len(pids)]
+			if pid := pids[rand.Int()%len(pids)]; pid != 0; {
```

Comment:
Hey @aranhaagency this won't work because the pid variable is out of scope. What's the behavior if it is 0? What you need to check if is the result of `rand.Int()%len(pids)` overflows the length of the `pids` list

Reply:
Thanks @rauljordan, so can i create a function to check if overflow occurs?

Reply:
I create this function checkPidOverflow()